calendarx.d.ts

```diff
@@ -0,0 +1,82 @@
+import * as React from "react";
+
+export type DateLike = Date | string | number
+
+export interface Event {
+    date: Date;
+}
+
+export type View = 'year' | 'month' | 'week' | 'day'
+
+export type ButtonPropsGetter = () => { label: string, onClick: () => void }
+
+export type Day = {
+    date: Date,
+    events: Event[],
+    isToday: boolean,
+    isSame: (view: View) => boolean
+}
+
+export type Properties = {
+    date: Date,
+    days: Day[][],
+    headers: string[],
+    view: View,
+    numDays: number,
+    jump: (n: number, v: View) => void,
+    goToNext: () => void,
+    goToPrev: () => void,
+    goToToday: () => void,
+    goToDate: (date: DateLike) => void,
+    getPrevButtonProps: ButtonPropsGetter,
+    getNextButtonProps: ButtonPropsGetter,
+    getTodayButtonProps: ButtonPropsGetter,
+    setNumDays: (numDays: number) => void
+}
+
+export interface Options {
+    initialDate?: DateLike;
+    initialNumDays?: number;
+    date?: DateLike;
+    numDays?: number,
+    events?: Event[];
+    weekStartsOn?: number;

```

Comment:
I think you can be a bit more specific here to avoid confusion around whether the  days are 0-indexed or not:
```
weekStartsOn?: 0 | 1 | 2 | 3 | 4 | 5 | 6
```